TASVideos.WikiEngine/NodeImplementations/Module.cs

```diff
@@ -29,9 +41,11 @@ public async Task WriteHtmlAsync(TextWriter w, WriterContext ctx)
 		{
 			if (!ctx.AddTdStyleFilter(Parameters))
 			{
-				var div = new Element(CharStart, "div") { CharEnd = CharEnd };
-				div.Children.Add(new Text(CharStart, "Module Error for settableattributes: Couldn't parse parameter string.") { CharEnd = CharEnd });
-				div.Attributes["class"] = "module-error";
+				Element div = new(
+					_charRange,
+					"div",
+					attributes: [new("class", "module-error")],
```

Comment:
This creates an extra object for the attributes collection compared to the old version of the code.  This isn't entirely a new problem, as there were some other callsites that wasted objects before, but it's worse now.

Reply:
~~Wasn't it allocating the same number of collections before, just in the `Element` ctor?~~ Ah I see, this is an array + `ToDictionary`. Should I go back to inserting after init, or should I add constructors which take a `Dictionary`?

Reply:
Passing a `Dictionary` seems good so long as we can make sure we never accidentally share those dictionaries (the passed object should be owned by the element that created it)

Reply:
Sorry, do you mean owned by the `Element` or owned by the call-site? Not that I know how to put either into code—mutation of `Element.Attributes` is relied on by a few call-sites, so it can't simply be changed to `ImmutableDictionary` for example.

Reply:
Owned by the element; I don't want to accidentally share a single Dictionary across multiple elements, as that would be too confusing for now.

Reply:
There are actually fewer mutations than I thought (must have confused it with `Children`), but even after encapsulating them, I can't think of a good way to enforce that invariant that's not just defensive copying.
I tried `[CallerArgumentExpression]` + checking for `new Dictionary` or `.ToDictionary(`, but while it may have worked, it also would have made every callsite verbose. Plus, half the callsites use the `Attr` method, so they would have to be `new Dictionary(new[] { Attr(..., ...) })` which I believe allocates no less than `new KVP<string, string>[] { Attr(..., ...) }.ToDictionary()`.
Unless you have any ideas, I'll go ahead with replacing `IEnumerable`s with `Dictionary`s, and it will just have to rely on humans not misusing references.